README.md

```diff
@@ -13,40 +13,72 @@
     <a href="https://github.com/featuretoggles/feature-toggles"><img alt="Github CI" src="https://github.com/featuretoggles/feature-toggles/workflows/Node%20CI/badge.svg"></a>
 </p>
 
-# Feature Toggling the best way (Build time)
+# Feature Toggles
 
-Currently, This is available as babel plugin.
+A _Babel plugin_ that helps remove _unfinished code_ from your JavaScript and Node Js application on _build time_. This is a configuration based feature toggling.
 
-> Are you struggling with the maintenance of an unfinished feature?
-> Are you tired of long-running branches?
-> Do you want to implement Continuous Delivery and Trunk Based Development?
+> ### The list of problems stopping you to productive, then you are at right place.
+>
+> 1. How to maintain both existing and unfinished feature?
+> 2. How to get rid of long-lived branches?
+> 3. How to manage single codebase used by multiple teams?
+> 4. How to maintain a production-ready master branch?
+
+**Problem statement example:**\
+You are one of the several teams working on a project where you might be seen below situations.
+
+1. **Unfinished code:**
+
+- Given you working on a feature and that will take several times before goto production. In general, You create a feature branch and put all new code in it and keep updating until you and your team think the feature is ready for production and ready for merge to master branch.
+- Now think about another aspect of it. Suppose you took this feature to build 4 weeks in a separate branch. In the meantime, another team member of your team also building other feature and merge them to master branch.
+- The problem started when your feature branch is few days or weeks behind while you try to merge the code you find merge conflicts multiple in files. And trust me which is worse then you think.
+- You created a feature branch to manage unfinished code but you end up merge hell.
+
+2. **External dependency:**
+
+- Another situation where you ready with your feature in a separate branch and the backend code is not yet ready or the business doesn't want to release the feature now. What will happen after 2 months or 4 months your team says lets release that feature, do you think that will be easy like merge the code to master and release.
+- Not sure about you but when I faced this issue, it freaked me out.
+- Let's analyze what happened to me.
+  - First I hit with merge conflict and it took me hell lot of time to resolve and also need to talk to the developer whose code causes the conflicts.
+  - Okay I passed the situational and rebase is successful, handed over to the testing team. The testing team gave me a list of the bug to fix. What torture and if your team facing this more often then it's not you the problem its just the process we adopt that is the problem.
+- Merge leads you to bugs and the code is separately built so automated test will not be able to handle that.
```

Comment:
Handling merge conflicts are error prone and your automated tests might not catch that all the time